src/session/user.rs

```diff
@@ -146,8 +147,65 @@ mod imp {
 
             let avatar = obj.avatar();
             let user_expression = gtk::ConstantExpression::new(obj);
-            let full_name_expression = super::User::full_name_expression(&user_expression);
-            full_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
+            let display_name_expression = super::User::display_name_expression(&user_expression);
+            display_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
+        }
+    }
+    pub fn time_ago(unix: i32) -> String {
+        let datetime_utc = glib::DateTime::new_now_utc().unwrap().to_unix();
+        let unix: i64 = unix.into();
+        let last_seen = glib::DateTime::from_unix_utc(unix)
+            .unwrap()
+            .to_local()
+            .unwrap();
+        let datetime_local = glib::DateTime::new_now_local().unwrap();
+
+        let diff = datetime_utc - unix;
+        let minutes = diff / 60;
+        let hours = minutes / 60;
+        if datetime_local.format("%Y %m %d").unwrap() == last_seen.format("%Y %m %d").unwrap() {
+            if hours == 0 {
+                if minutes < 0 {
+                    gettext("just now")
+                } else if minutes == 1 {
+                    gettext("a minute ago")
+                } else {
+                    format!("{} {}", gettext("minutes ago"), minutes)
```

Comment:
It's wrong to assume positional arguments here since there're languages (e.g. Spanish in this case) that need a different word positioning. Same thing for the other string below.

Reply:
Right, please don't go even deeper I made it even worse (user.rs at line 411)

Reply:
telegrand:
![photo](https://user-images.githubusercontent.com/58047802/148975265-17216947-bd1b-404a-ac39-b25c753aaffa.png)
telegram (user subtitle):
![photo](https://user-images.githubusercontent.com/58047802/148975357-de245ece-afdb-49bd-bc7c-687fc7d7acd0.png)
telegram (user dialog):
![photo](https://user-images.githubusercontent.com/58047802/148975374-242a35c8-095f-47a6-9f42-4c7e1945ac9f.png)

Reply:
The question is, how far do we want to copy telegram IMO telegram implementation is a little inconsistent.